src/napari/_qt/layer_controls/widgets/qt_autoscale_control.py

```diff
@@ -0,0 +1,75 @@
+from typing import TYPE_CHECKING, Optional
+
+from qtpy.QtCore import Qt
+from qtpy.QtWidgets import QHBoxLayout, QPushButton, QWidget
+
+from napari._qt.layer_controls.widgets.qt_widget_controls_base import (
+    QtWidgetControlsBase,
+    QtWrappedLabel,
+)
+from napari.layers.base.base import Layer
+from napari.utils.events.event_utils import connect_no_arg, connect_setattr
+from napari.utils.translations import trans
+
+if TYPE_CHECKING:
+    from napari.layers import Image
+
+
+class AutoScaleButtons(QWidget):
+    def __init__(self, layer: 'Image', parent=None) -> None:
+        super().__init__(parent=parent)
+
+        self.setLayout(QHBoxLayout())
+        self.layout().setSpacing(2)
+        self.layout().setContentsMargins(0, 0, 0, 0)
+        once_btn = QPushButton(trans._('once'))
+        once_btn.setFocusPolicy(Qt.FocusPolicy.NoFocus)
+
+        auto_btn = QPushButton(trans._('continuous'))
+        auto_btn.setCheckable(True)
+        auto_btn.setFocusPolicy(Qt.FocusPolicy.NoFocus)
+        once_btn.clicked.connect(lambda: auto_btn.setChecked(False))
+        connect_no_arg(once_btn.clicked, layer, 'reset_contrast_limits')
+        connect_setattr(auto_btn.toggled, layer, '_keep_auto_contrast')
+        connect_no_arg(auto_btn.clicked, layer, 'reset_contrast_limits')
+
+        self.layout().addWidget(once_btn)
+        self.layout().addWidget(auto_btn)
+
+        # just for testing
+        self._once_btn = once_btn
+        self._auto_btn = auto_btn
+
+
+class QtAutoScaleControl(QtWidgetControlsBase):
+    """
+    Class that wraps the connection of events/signals between the layer auto-contrast related
```

Comment:
Any reason why this is separate from clim controls? Shouldn't they always be together?

Reply:
This comes from the `AutoScaleButtons` class definition being independent from the contrats limit related widget in `main` https://github.com/napari/napari/blob/deb2871c96b4997e2cc165e665241dcf51d41c3e/src/napari/_qt/layer_controls/qt_image_controls_base.py#L246

But maybe a single wrapper class should have both widgets (contrats limits slider + autoscale buttons)?

Reply:
Yeah that's what I was thinking